examples/example_pandas_train_test.py

```diff
@@ -0,0 +1,124 @@
+# -*- encoding: utf-8 -*-
+"""
+===============================
+Test and Train data with Pandas
+===============================
+
+*auto-sklearn* can automatically encode categorical columns using a label/ordinal encoder.
+This example highlights how to properly set the dtype in a DataFrame for this to happen,
+and showcase how to input also testing data to autosklearn.
+The X_train/y_train arguments to the fit function will be used to fit the scikit-learn model,
+whereas the X_test/y_test will be used to evaluate how good this scikit-learn model generalizes
+to unseen data (i.e. data not in X_train/y_train). Using test data is a good mechanism to measure
+if the trained model suffers from overfit, and more details can be found on `evaluating estimator
+performance <https://scikit-learn.org/stable/modules/cross_validation.html#cross-validation>`_.
+This example further highlights through a plot, the best individual models found by *auto-sklearn*
+through time (under indv_model_train_score/indv_model_test_score's legend). It also shows the
+training and test performance of the ensemble build using the best performing models (under
+ensemble_train_score and ensemble_test_score respectively).
+"""
+import time
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import sklearn.model_selection
+import sklearn.datasets
+import sklearn.metrics
+
+import autosklearn.classification
+from autosklearn.metrics import accuracy
+
+
+def get_runhistory_models_performance(automl):
+    metric = cls._automl[0]._metric
+    data = automl._automl[0].runhistory_.data
+    performance_list = []
+    for run_key, run_value in data.items():
+        if 'train_loss' not in run_value.additional_info:
+            # Ignore crashed runs
+            continue
+        endtime = pd.Timestamp(time.strftime('%Y-%m-%d %H:%M:%S',
+                                             time.localtime(run_value.endtime)))
+        train_score = metric._optimum - (metric._sign * run_value.additional_info['train_loss'])
+        test_score = metric._optimum - (metric._sign * run_value.additional_info['test_loss'])
+        performance_list.append({
+            'Timestamp': endtime,
+            'indv_model_train_score': train_score,
```

Comment:
Sorry, I guess I said training loss, but actually we'd be interested in the validation loss as that's what we're optimizing.

Reply:
oh, I see, I will use the cost directly then

Reply:
Could you please also rename this to validation score?

Reply:
I think we should also rename the ensemble to be validation.

But I found a legacy code that we need to discuss. Y_valid in the ensemble builder doesn't make sense, as it relies on datamanager to have x_valid. This is no longer the case right?

I think this should be a separate PR, to prune up this Y_valid.

For this PR I will just rename the column. Let me know if this makes sense

Reply:
> I think we should also rename the ensemble to be validation.

Yes and no. There it's actually the training performance and not the validation performance. 

> But I found a legacy code that we need to discuss. Y_valid in the ensemble builder doesn't make sense, as it relies on datamanager to have x_valid. This is no longer the case right?

Yes, I think you can remove all the logic here.

> For this PR I will just rename the column. Let me know if this makes sense

Okay. How about calling the strings 'single_best_optimization_score', 'single_best_test_score', 'ensemble_optimization_score' and 'ensemble_test_score'?

Reply:
thanks for the suggestion, I will go with this naming